### PR TITLE
WP-1813 replace ft scroller

### PR DESCRIPTION
**Pull request by jciolek**
A few notes:

- I have removed the FTScroller completely and replaced it with a relatively simple React component.
- I have simplified the implementation of the Carousel, making it slightly easier to follow.
- I have fixed issues around resizing, which were infesting the Carousel.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,13 +97,15 @@
   "stylelint": {
     "extends": [
       "stylelint-config-strict"
-    ]
+    ],
+    "rules": {
+      "no-indistinguishable-colors": null
```
Comment (fniewijk) [inline package.json]: Did you have problems with the linter?

File: src/index.js [hunk 2]
```diff
@@ -204,123 +129,89 @@ export default class Carousel extends React.Component {
...
-    // The noscript tag causes react to fail and so have to render on the server.
-    const noScript = typeof window === 'undefined' ?
-      (
-        <noscript>
-          {/* eslint-disable react/no-danger  */}
-          <style dangerouslySetInnerHTML={{ __html: styles }} />
-          {/* eslint-enable react/no-danger  */}
-        </noscript>
-      ) :
-      null;
```
Comment (fniewijk) [inline src/index.js]: Yuk. Did you test the improved carousel with javascript disabled? Is that still important?

File: src/parts/carousel-scroller.js [hunk 3]
```diff
@@ -0,0 +1,235 @@
+import React from 'react';
```
Comment (fniewijk) [inline src/parts/carousel-scroller.js]: Needs tests

File: package.json (hunk 1 above)
Comment (jciolek) [reply package.json]: I have ported the ftscroller CSS to the carousel and this rule (which is deprecated, btw) raised its ugly head. It's about colours being too similar to each other, which in my view did not make an awful lot of sense in this case.

File: src/index.js (hunk 2 above)
Comment (jciolek) [reply src/index.js]: I was hoping the comment above the replaced code would explain it.

File: src/parts/carousel-scroller.js (hunk 3 above)
Comment (jciolek) [reply src/parts/carousel-scroller.js]: I've added tests reflecting the new API.

File: src/index.js (hunk 2 above)
Comment (fniewijk) [reply src/index.js]: I think I didn't ask my question well enough. Should the component support graceful degradation for browsers with javascript turned off?

Comment (jciolek) [reply src/index.js]: No, I got your question. That's exactly what it is. With JS switched off there is only SSR, so this works. It's an equivalent of the previous solution, which was horrible.

**Pull request merged**